hibernate-core/src/main/java/org/hibernate/resource/transaction/backend/jdbc/internal/JdbcResourceLocalTransactionCoordinatorImpl.java

```diff
@@ -84,8 +84,7 @@ public void explicitJoin() {
 
 	@Override
 	public boolean isJoined() {
-		return physicalTransactionDelegate != null && physicalTransactionDelegate.getStatus() == TransactionStatus.ACTIVE;
-
+		return physicalTransactionDelegate != null && getTransactionDriverControl().isActive( true );
```

Comment:
Not sure I like this.  So if an EntityManager explicitly joins a transaction (`EntityManager#joinTransaction`), after rollback-only is set we will report that the transaction is no longer joined (`EntityManager#isJoinedToTransaction`)?  That does not seem right to me.